### PR TITLE
chore: update mail ADR

### PR DESCRIPTION
**Pull request by jjtang1985**
This PR does the following:
- ADR updates
- clean up comments
- update tests

Closes SAP/cloud-sdk-backlog#ISSUENUMBER.

### CODE REVIEW

File: knowledge-base/adr/0033-send-email.md [hunk 1]
```diff
@@ -145,3 +146,17 @@ Applications on BTP can send mails with `MAIL` destinations to cloud/On-Prem mai
...
+  Working with `nodemailer` + basic auth
+- using socket protocol
+  Our PoC tried `WS.WebSocket`, but as socket protocol without knowing SMTP, it did not support email related APIs.
+  E.g, authentication + email properties (from/to...) cannot be configured
+#### OnPrm Proxy
+- support socket proxy for now
+  It uses custom auth handler with JWT.
+- http proxy (later)
+  Below is one example with `node-http-proxy-agent`, that is not tested, and has cumbersome implementation.
+  [Useful example](https://github.com/TooTallNate/node-http-proxy-agent/blob/master/src/agent.ts#L83)
```
Comment (FrankEssenberger) [inline knowledge-base/adr/0033-send-email.md:162]: [nth] The inclusive plugin is complaining about the 'master' in the path. Perhaps you can somehow ignore this chec.

**Review by FrankEssenberger — APPROVED** [review]

File: test-packages/integration-tests/test/auth-flows/auth-flow.spec.ts [hunk 2]
```diff
@@ -506,15 +488,12 @@ describe('OAuth flows', () => {
...
-    // TODO: use the key `onpremise_socks5_proxy_port` for the proxy type of the email destination instead of `onpremise_proxy_port` (normal OnPrem HTTP destination)
-    // The test below should pass now, as the port is anyway a number.
-    // Please double check the value, before enabling.
-    // expect(destination!.proxyConfiguration).toMatchObject({
-    //   headers: { 'Proxy-Authorization': expect.stringMatching(/Bearer.*/) },
-    //   host: expect.stringMatching(/.*sap\.hana\.ondemand\.com/),
-    //   port: expect.stringMatching(/\d+/),
-    //   protocol: 'http'
-    // });
+    expect(destination!.proxyConfiguration).toMatchObject({
```
Comment (marikaner) [inline test-packages/integration-tests/test/auth-flows/auth-flow.spec.ts:491]:
```suggestion
    expect(destination?.proxyConfiguration).toMatchObject({
```

File: knowledge-base/adr/0033-send-email.md [hunk 3]
```diff
@@ -145,3 +146,17 @@ Applications on BTP can send mails with `MAIL` destinations to cloud/On-Prem mai
...
 - [Using socks proxy with nodemailer](https://nodemailer.com/smtp/proxies/#2-using-socks-proxy)
+
+### PoC notes
+#### Protocol for sending emails
+- using SMTP protocol
+  Working with `nodemailer` + basic auth
+- using socket protocol
+  Our PoC tried `WS.WebSocket`, but as socket protocol without knowing SMTP, it did not support email related APIs.
+  E.g, authentication + email properties (from/to...) cannot be configured
+#### OnPrm Proxy
```
Comment (marikaner) [inline knowledge-base/adr/0033-send-email.md:157]:
```suggestion
#### OnPrem Proxy
```

**Review by marikaner — COMMENTED** [review]
Just left two minor suggestions. I wish the PoC notes were cleaned up a tiny bit more (lists aligned, formatting corrected, etc.).

**Pull request merged**